temporal_capi/src/calendar.rs

```diff
@@ -0,0 +1,61 @@
+#[diplomat::bridge]
+#[diplomat::abi_rename = "temporal_rs_{0}"]
+#[diplomat::attr(auto, namespace = "temporal_rs")]
+pub mod ffi {
+    use crate::error::ffi::TemporalError;
+    use diplomat_runtime::DiplomatStr;
+
+    #[diplomat::enum_convert(icu_calendar::any_calendar::AnyCalendarKind, needs_wildcard)]
+    pub enum AnyCalendarKind {
+        Buddhist,
+        Chinese,
+        Coptic,
+        Dangi,
+        Ethiopian,
+        EthiopianAmeteAlem,
+        Gregorian,
+        Hebrew,
+        Indian,
+        IslamicCivil,
+        IslamicObservational,
+        IslamicTabular,
+        IslamicUmmAlQura,
+        Iso,
+        Japanese,
+        JapaneseExtended,
+        Persian,
+        Roc,
+    }
```

Comment:
Are there tests you can setup to make sure the exposed enums match up with the underlying enum data?

Edit: After reading [this](https://github.com/boa-dev/temporal/issues/104#issuecomment-2521751143) it sounds like ICU4X already has such scripts to ensure this in CI. I guess down the line we will want to add the same

Reply:
The ICU4X script doesn't handle variants. We can enable the [`non_exhaustive_omitted_patterns`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#non-exhaustive-omitted-patterns) lint.